moveit_simple/test/tool_windup.cpp

```diff
@@ -0,0 +1,149 @@
+/*
+ * Software License Agreement (Apache License)
+ *
+ * Copyright (c) 2019 Plus One Robotics
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <eigen_conversions/eigen_msg.h>
+#include <gtest/gtest.h>
+#include <moveit_simple/moveit_simple.h>
+#include <moveit_simple/prettyprint.hpp>
+#include <random_numbers/random_numbers.h>
+
+using testing::Types;
+
+namespace moveit_simple_test
+{
+
+/**
+ * @brief UserRobotTest is a fixture for testing public methods.
+ * Objects that can be directly used inside the test
+ - robot pointer for moveit_simple::OnlineRobot
+*/
+class UserRobotTest : public ::testing::Test
+{
+protected:
+  std::unique_ptr<moveit_simple::OnlineRobot> robot;
+  virtual void SetUp()
+  {
+  robot = std::unique_ptr<moveit_simple::OnlineRobot> (new moveit_simple::OnlineRobot
+                  (ros::NodeHandle(), "robot_description", "manipulator"));
+  ros::Duration(2.0).sleep();  //wait for tf tree to populate
+  }
+  virtual void TearDown()
+  {}
+};
+
+/**
+ * @brief DeveloperRobot is a class inheriting from moveit_simple::OnlineRobot to test protected methods
+ * Add the protected methods(that are required to be tested) in the following list
+*/
+class DeveloperRobot: public moveit_simple::OnlineRobot
+{
+public:
+  using moveit_simple::OnlineRobot::OnlineRobot;
+  using moveit_simple::OnlineRobot::addTrajPoint;
+  using moveit_simple::OnlineRobot::toJointTrajectory;
+  using moveit_simple::OnlineRobot::interpolate;
+  using moveit_simple::OnlineRobot::jointInterpolation;
+  using moveit_simple::OnlineRobot::cartesianInterpolation;
+  using moveit_simple::OnlineRobot::isInCollision;
+};
+
+/**
+ * @brief DeveloperRobotTest is a fixture for testing protected methods.
+ * Objects that can be directly used inside the test
+ - robot pointer for DeveloperRobot
+*/
+
+class DeveloperRobotTest : public ::testing::Test
+{
+protected:
+  std::unique_ptr<DeveloperRobot> robot2;
```

Comment:
```suggestion
  std::unique_ptr<DeveloperRobot> robot_;
```
why 2?

Reply:
That's taken from other test sources where there is a `UserRobot` "robot" and a `DeveloperRobot` "robot2"